CommonItem/src/main/java/com/github/sef24sp4/common/item/lootTable/LootTable.java

```diff
@@ -0,0 +1,93 @@
+package com.github.sef24sp4.common.item.lootTable;
+
+import com.github.sef24sp4.common.item.CommonItem;
+import com.github.sef24sp4.common.item.ItemRarity;
+import com.github.sef24sp4.common.item.ItemSPI;
+
+import java.util.*;
+
+public class LootTable {
+	private final Map<ItemRarity, Double> itemChances;
+	private final Collection<ItemSPI> itemProviders;
```

Comment:
Might be relevant to replace this with:
```java
private final Map<ItemRarity, Collection<ItemSPI>> itemProviders;

/*
 * then call in constructor
 * this.itemProviders = generateItemProvidersMap(...);
 */

private static Map<ItemRarity, Collection<ItemSPI>> generateItemProvidersMap(final Collection<ItemSPI> providers) {
	final Map<ItemRarity, Collection<ItemSPI>> map = new HashMap<>();
	for (ItemSPI provider : providers) {
		map.computeIfAbsent(provider.getRarity(), k -> new ArrayList<>()).add(provider);
	}
	return Collections.unmodifiableMap(map);
}
```

Can then simplify `getItem` method. Also greatly reduces computations, as provides are only iterated through once.

Reply:
Not sure if I implemented it correctly, but all other requests should be fixed :)